app/src/main/java/com/macbitsgoa/ard/activities/helpers/AuthHelper.java

```diff
@@ -0,0 +1,155 @@
+import com.google.android.gms.auth.api.signin.GoogleSignInResult;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthCredential;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.auth.GoogleAuthProvider;
+import com.google.firebase.database.DatabaseReference;
+import com.macbitsgoa.ard.R;
+import com.macbitsgoa.ard.activities.AuthActivity;
+import com.macbitsgoa.ard.utils.AHC;
+
+import static com.macbitsgoa.ard.utils.AHC.FDR_USERS_EMAIL;
+import static com.macbitsgoa.ard.utils.AHC.FDR_USERS_NAME;
+import static com.macbitsgoa.ard.utils.AHC.FDR_USERS_PHOTO_URL;
+
+/**
+ * Helper class to handle backend functions for {@link AuthActivity}.
+ * @author Rushikesh Jogdand
+ */
+@SuppressWarnings("WeakerAccess")
+public class AuthHelper
+        implements GoogleApiClient.OnConnectionFailedListener {
+
+    /**
+     * Tag for this class.
+     */
+    private static final String TAG = "MAC_ARD.AuthHelper";
+
+    /**
+     * Do when signed in to firebase with credential provided by google.
+     */
+    public OnCompleteListener<AuthResult> googleAuthCompleteListener = new OnCompleteListener<AuthResult>() {
+        @Override
+        public void onComplete(@NonNull final Task<AuthResult> task) {
+            if (!task.isSuccessful()) {
+                Log.e(TAG, "authWithGoogle-onComplete: task unsuccessful", task.getException());
+                hostActivity.handleGoogleSignInFailure();
+                return;
+            }
+            updateSelfInfo();
+            hostActivity.finish();
+        }
+    };
+
+    /**
+     * Calling Activity.
+     */
+    private AuthActivity hostActivity;
+
+    public AuthHelper(final AuthActivity hostActivity) {
+        this.hostActivity = hostActivity;
+    }
+
+    /**
+     * Get FirebaseAuth instance.
+     * @return {@link FirebaseAuth} instance.
+     */
+    protected FirebaseAuth getFirebaseAuth() {
+        return FirebaseAuth.getInstance();
+    }
+
+    /**
+     * Send relevant info of current user to firebase.
+     */
+    public void updateSelfInfo() {
+        final FirebaseUser user = getFirebaseAuth().getCurrentUser();
+        if (user == null) {
+            return;
+        }
+        final String uid = user.getUid();
+        final String name = user.getDisplayName();
+        final String email = user.getEmail();
+        String photoUrl = null;
+        final Uri uri = user.getPhotoUrl();
+        if (uri != null) {
+            photoUrl = uri.toString();
+        }
+        final DatabaseReference userDb = hostActivity.getRootReference().child(AHC.FDR_USERS).child(uid);
+        if (name != null && name.length() > 0) {
```

Comment:
are length checks required if data is not null?
Even if they are of length 0, we need to put some placeholder right? (that too something like `""` which also has length 0?)